[tune] MedianStopping on result

waldroje opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

## What do these changes do?
Added a class MedianStoppingResult to schedulers, to provide median stopping of trials as they are being run, similar to current implementation of PBT, rather than waiting for trials to complete.


## Related issue number

<!-- For example: "Closes #1234" -->

Closes #5335

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
I tried to run linter script, but kept getting error related to ci/travis/format.sh.. not something i at all familar with.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16118/
Test PASSed.

richardliaw commented:
I've run the linter and have merged this with MedianStoppingRule. 

I'll do a closer examination and help add some tests tomorrow.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16180/
Test PASSed.

richardliaw commented:
Hi @waldroje, I made some changes to the PR. I made the comparison logic a bit more isolated (so that `best` values were not flipped by `-1`), and I removed the roundrobin logic (the pausing). This is because pausing can be extraordinarily expensive especially when dealing with neural networks. Let me know if that's a feature that you need and we can look into putting it back (probably under a feature flag).

I'll need to modify tests (in `tune/tests/test_trial_scheduler`), but if you want to take a look into that, feel free.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16210/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16214/
Test PASSed.

waldroje commented:
Richard,
Thanks for cleaning the code up... can you elaborate a bit on what you mean by "expensive"? Is it just in wasted i/o time? Or is it also difficult to start / stop trials cleanly? 

I will say in practice, I am tending to agree with you, as clearly the overhead must be extensive, as such regular starting/stopping slows down each individual trial by huge factor in my limited experience running this version. I have not really tested extending the iteration time, to allow each individual trial to run longer, to see the relative impact...

What I was trying to achieve initially was a way to scale up APEX style runs simultaneously, as in those cases you are limited in your ability to scale the number of trials by the number of workers available on the head node, in terms of allocating drivers/shards... as opposed to something like DQN where you can scale up almost with no limit.  If I can't pause trials, the number of trials is effectively limited to the available resources, which for APEX type runs, is usually not much, as a standard p3.2xlarge has 1 gpu, 8 cpu... so in theory the most one could split the gpu is to 0.125, 1 driver per cpu, but my experience is that type of run usually fails quickly, I assume due to the resource consumption of the replay shards... even splitting to 0.25 and setting replay_shards to 1, I tend to avoid, as it leaves no headroom, and obviously performance suffers... so that leaves 0.50, and thus only 2 trials can be run at a time, thus using your implementation, I can only compare 2 trials to each other at a time, which doesn't really provide much gain over running independently.

My experience has been that organizing parameter sweeps is really hard, and I have found over time I do a poor job of it and often repeat trials, or run trials with little difference in parameter space... in my head I felt it might be worth to be able to launch what seems like an expensive run, yet one that consolidates everything one place, with clear comparisons, and might in the long run actually save time/resources. But it's only worth it if the run can stay up and not fail, as well as not add unreasonable overhead, and in this case I felt it wasn't any more than PBT, as over time resources would be shut down as trials terminate, which might balance out against the more frequent start/stops...

Let me know your thoughts, and if anything I've written there doesn't make sense or is incorrect.

richardliaw commented:
Thanks for the thorough reply! This is helpful in understanding your use
case and decisions. Replies inline:

> Thanks for cleaning the code up... can you elaborate a bit on what you
mean by "expensive"? Is it just in wasted i/o time? Or is it also difficult
to start / stop trials cleanly?
By this, I mean that serialization overhead can be quite costly - if the
iteration takes 1 second and overhead of serialization/deserialization of
the policy/replay buffer is 2 seconds, round robin-ing among multiple
trials can make the entire experiment take significantly longer.

> If I can't pause trials, the number of trials is effectively limited to
the available resources, which for APEX type runs, is usually not much, as
a standard p3.2xlarge has 1 gpu, 8 cpu...
Have you looked at other instance types? If you need V100, what happens if
you just add another p3.2xlarge to your cluster? (See below if these are
not options).

> 2 trials can be run at a time, thus using your implementation, I can only
compare 2 trials to each other at a time, which doesn't really provide much
gain over running independently.
If your machine can only run 2 at a time, one possibility for you is to
look at HyperBand (not the asynchronous one)
<https://ray.readthedocs.io/en/latest/tune-schedulers.html#hyperband>.
It'll round-robin the trials, terminate unpromising ones until you have one
left. Another option is to add a `round_robin_freq` flag that will pause
the execution every X iterations.

Let me know your thoughts!

On Sun, Aug 11, 2019 at 5:37 AM waldroje <notifications@github.com> wrote:

> Richard,
> Thanks for cleaning the code up... can you elaborate a bit on what you
> mean by "expensive"? Is it just in wasted i/o time? Or is it also difficult
> to start / stop trials cleanly?
>
> I will say in practice, I am tending to agree with you, as clearly the
> overhead must be extensive, as such regular starting/stopping slows down
> each individual trial by huge factor in my limited experience running this
> version. I have not really tested extending the iteration time, to allow
> each individual trial to run longer, to see the relative impact...
>
> What I was trying to achieve initially was a way to scale up APEX style
> runs simultaneously, as in those cases you are limited in your ability to
> scale the number of trials by the number of workers available on the head
> node, in terms of allocating drivers/shards... as opposed to something like
> DQN where you can scale up almost with no limit. If I can't pause trials,
> the number of trials is effectively limited to the available resources,
> which for APEX type runs, is usually not much, as a standard p3.2xlarge has
> 1 gpu, 8 cpu... so in theory the most one could split the gpu is to 0.125,
> 1 driver per cpu, but my experience is that type of run usually fails
> quickly, I assume due to the resource consumption of the replay shards...
> even splitting to 0.25 and setting replay_shards to 1, I tend to avoid, as
> it leaves no headroom, and obviously performance suffers... so that leaves
> 0.50, and thus only 2 trials can be run at a time, thus using your
> implementation, I can only compare 2 trials to each other at a time, which
> doesn't really provide much gain over running independently.
>
> My experience has been that organizing parameter sweeps is really hard,
> and I have found over time I do a poor job of it and often repeat trials,
> or run trials with little difference in parameter space... in my head I
> felt it might be worth to be able to launch what seems like an expensive
> run, yet one that consolidates everything one place, with clear
> comparisons, and might in the long run actually save time/resources. But
> it's only worth it if the run can stay up and not fail, as well as not add
> unreasonable overhead, and in this case I felt it wasn't any more than PBT,
> as over time resources would be shut down as trials terminate, which might
> balance out against the more frequent start/stops...
>
> Let me know your thoughts, and if anything I've written there doesn't make
> sense or is incorrect.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ray-project/ray/pull/5402?email_source=notifications&email_token=ABCRZZIXTKOOIL2FUNPQGX3QEABZTA5CNFSM4IKFGW42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4BABVY#issuecomment-520224983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCRZZK3FX5YTL5R77ZIYIDQEABZTANCNFSM4IKFGW4Q>
> .
>

waldroje commented:
Have you looked at other instance types? If you need V100, what happens if
you just add another p3.2xlarge to your cluster? (See below if these are
not options).

- as a worker? My understanding of cluster setup is there is a head, and workers, and all the workers need to be the same type... so if I add another p3.2xlarge, then I will have 2 gpus, but only 16 cpus.. and to run APEX I would like at least 16-32 workers per trial... i normally add 1-2 c5.9xlarge with 36 cpu resources each to a p3.2xlarge.

Another option is to add a `round_robin_freq` flag that will pause
the execution every X iterations.

- that is what the "eval_interval" parameter is supposed to do... i.e. results are collected every iteration, but you don't actually pause/switch trials each iteration, you only do that at each 'eval_interval", which in my planning, the time factor was "training_iteration", so I can raise that and thus allow each trial to advance in multiple iteration steps prior to being paused/evaluated.

If your machine can only run 2 at a time, one possibility for you is to
look at HyperBand (not the asynchronous one)

- I did take a quick look at that, and I can't recall exactly why I moved on to the MedianStoppingRule.. but I believe it might have failed in my early attempts.. it's possible I overloaded it though, so I may take another look at that with a better understanding of the resource issues.

Last bit of additional info, is my custom env is pretty slow, using my normal params, an iteration often takes around 60s... so if i'm only evaluating every 4-5 iterations, i'm actually running for 240-300s vs. the 2s for serialization, but that said, i definitely recognize the cost and have observed it in running some trials this way.

waldroje commented:
Richard,

I took another shot at Hyperband per your suggestion, and was able to get it to successfully run a large batch of trials... I started a parameter search which began with about 48 different trials... it created 4 brackets, and after running overnight there were a total of 5 trials still running, 23 had terminated, and 20 had errored... I did not check them all but the ones I checked all had errored at the same spot...

```
Traceback (most recent call last):
  File "/home/ubuntu/algo/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 520, in _process_trial
    self, trial, result)
  File "/home/ubuntu/algo/lib/python3.6/site-packages/ray/tune/schedulers/hyperband.py", line 171, in on_trial_result
    bracket.update_trial_stats(trial, result)
  File "/home/ubuntu/algo/lib/python3.6/site-packages/ray/tune/schedulers/hyperband.py", line 372, in update_trial_stats
    assert delta >= 0
AssertionError
```

My time_attr is "training_iteration", and from the code it seems that a result returned the same iteration as the previous one? How is that possible? The comments in the method "update_trial_stats" mention 
```
Update result for trial. Called after trial has finished
        an iteration - will decrement iteration count.
```
but the code shows an increment... is that the way it is supposed to be? or is the comment just incorrect?
```
        assert delta >= 0
        self._completed_progress += delta
```

waldroje commented:
Per the above, it looks like I got that wrong, seems there is usually a first error, which is an actor died unexpectedly, and when I look at the plasma output, it looks like maybe objects are being evicted, so probably just needs more memory for the object store... and then the above error is triggered...

richardliaw commented:
Sorry for the slow reply - 

> Last bit of additional info, is my custom env is pretty slow, using my normal params, an iteration often takes around 60s... so if i'm only evaluating every 4-5 iterations, i'm actually running for 240-300s vs. the 2s for serialization, but that said, i definitely recognize the cost and have observed it in running some trials this way.

OK makes sense, we can probably just flag it off but include it.

> but the code shows an increment... is that the way it is supposed to be? or is the comment just incorrect?

> Per the above, it looks like I got that wrong, seems there is usually a first error, which is an actor died unexpectedly, and when I look at the plasma output, it looks like maybe objects are being evicted, so probably just needs more memory for the object store... and then the above error is triggered...

OK make sense, I don't think HyperBand is rigorously tested under cases of error, so maybe that check is not necessary.

waldroje commented:
No problem, I appreciate you spending so much time on something that is probably not that important for most... I realized overnight that the MedianStoppingResult version I had produced won't work as planned anyway, as I had omitted the actual checkpointing, so anything that is Paused, currently just goes back and restarts... so I need to implement the actual checkpoint process similar to PBT....

I did get a Hyperband run to finish.. although one trial had the error I listed above... in the actual experiment path the only errors listed were that one, but when I looked in the session logs, I did see reference to objects being evicted again.. even though I had bumped object store to 40gb... it's something I'll continue to look at, but my main issue is a single trial with my env generally takes between 24-48 hrs to finish, and consumes most of the resources of my head node... so it's hard to envision running a hyperparameter sweep and having to let a couple trials run all the way to finish, which is the case in both MedianStoppingRule as well as HyperBand, rather than having some process to eliminate poor runs in a concurrent fashion.

richardliaw commented:
> No problem, I appreciate you spending so much time on something that is probably not that important for most... I realized overnight that the MedianStoppingResult version I had produced won't work as planned anyway, as I had omitted the actual checkpointing, so anything that is Paused, currently just goes back and restarts... so I need to implement the actual checkpoint process similar to PBT....

Are you sure? This should just work - unless somehow you're extending RLlib in a way that doesn't implement chekcpointing?

> I did get a Hyperband run to finish.. although one trial had the error I listed above... in the actual experiment path the only errors listed were that one, but when I looked in the session logs, I did see reference to objects being evicted again.. even though I had bumped object store to 40gb... it's something I'll continue to look at, but my main issue is a single trial with my env generally takes between 24-48 hrs to finish, and consumes most of the resources of my head node... so it's hard to envision running a hyperparameter sweep and having to let a couple trials run all the way to finish, which is the case in both MedianStoppingRule as well as HyperBand, rather than having some process to eliminate poor runs in a concurrent fashion.

Another option is just to utilize experiment-level checkpointing along with top-k (keep_checkpoint_num). That should technically make your experiment resilient to failure. Well, does HyperBand mainly work for you then?

waldroje commented:
Yes, I think Hyperband works, and I will probably use that a bit...

In regards to the first point, once I put in the the following line of code, similar to PBT, everything worked as expected..

```
            state.last_checkpoint = trial_runner.trial_executor.save(
                trial, Checkpoint.MEMORY)   
```
If in tune.run you submit an experiment with a specific checkpoint frequency, like every 25 iterations, but then have a scheduler like I put in that pauses at some frequency less than the checkpoint_frequency, wouldn't you need the extra code above? What would automatically make it checkpoint? I'm simply doing a tune.run ... and using the scheduler and standard search_algo.. but I'm copying a config that has defined checkpoint frequency in it, which is longer than the evaluation frequency I had defined in the MedianStopping.

richardliaw commented:
Hm, really? I think the `PAUSE` code should be able to handle the checkpoint transparently without you manually calling `save(trial, MEMORY)`. 

The Tune checkpointing frequency automatically checkpoints to disk, but PAUSING and RESUMING from paused state should work...

In any case, I actually think what's in this PR is quite useful and would be happy to get it merged (though there are some tests failing). Could you help me take a look at them?

waldroje commented:
Yes, I did notice that when I got looking into the Trial code, that PAUSE itself initiates a save, so I need to take a close look and see if there was something else causing what I was seeing… I’d definitely like to help get some version of this in, as I do find it useful, let me know what the biggest issue is, and I’ll do what I can to help get it cleaned up.



On Aug 14, 2019, at 10:03 PM, Richard Liaw <notifications@github.com> wrote:

Hm, really? I think the PAUSE code should be able to handle the checkpoint transparently without you manually calling save(trial, MEMORY).

The Tune checkpointing frequency automatically checkpoints to disk, but PAUSING and RESUMING from paused state should work...

In any case, I actually think what's in this PR is quite useful and would be happy to get it merged (though there are some tests failing). Could you help me take a look at them?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/ray-project/ray/pull/5402?email_source=notifications&email_token=AHSHWMI7BZPAZJ34F3YAY73QES2O3A5CNFSM4IKFGW42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4KT72I#issuecomment-521486313>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AHSHWMLFK2MNAVYEGFMBISLQES2O3ANCNFSM4IKFGW4Q>.

richardliaw commented:
Basically, some of the MedianStoppingRule tests in `tests/test_trial_scheduler` are failing: https://travis-ci.com/ray-project/ray/jobs/224167914

Due to the algorithm changing, it's obvious that some of the tests need to change. If you could take a look into these tests and adjust them as necessary, I would really appreciate it!

waldroje commented:
So I made a few changes again, copied in the changes you made to some of the parameter names/types... I also merged the median result, / best_result calls into a single method, as didn't make sense to me to repeat the steps... I have left in the pbt style PAUSE, but added a resource check.. in addition i added a file to tune/tests/test_median_stopping_result.py ... which works, and generally can replace what is currently in the test suite in my opinion.  There are probably some additional tests that could be inserted at some point.  The file changes blew up a bit because I fetched and rebased, so hopefully that's not a big pain... in reality the only changes are in the 3 files, __init__, median_stopping_result, and now the test file.

richardliaw commented:
Can you do like `git pull . master`?

waldroje commented:
Says already up to date when I do that

waldroje commented:
one thing I noticed, something that might need to be reconsidered, is that if there are no resource constraints, and all trials are running in parallel, the trials that get out ahead, for various reasons, in theory have no actual peer to compare against... i..e t_max parameter ensures you don't compare trails which are way ahead in time, but there is also no mechanism to compare all trials at a similar point in time... it seems what might really make sense is to wait until all the trials have gotten to some eval increment, and then compare all the results at that point, and terminate all under-performing trials, irregardless of which one triggered the check via on_trial_result.. When the resources are constrained, and it's pausing, they will tend to stay in sync, or at least close...

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16317/
Test PASSed.

waldroje commented:
Sorry for all the commits, still have not figured out how to rebase properly.. I added another scheduler, I didn't run tests, wanted to see if it was something you were interested in first... I think it's the better version, as it evaluates all running trials at the same spot, and can stop a trial even if it is far ahead of slower trials based on past performance... it only evaluates once all have reached a multiple of the evaluation interval, and then looks at results in that window., so rather than evaluating each trial all the time, evaluates all running trials at the same time on a periodic schedule. This is the one I think I'm going to use, and if you think it's worthwhile, I'll fix the tests up.

richardliaw commented:
I usually use
git checkout master
git pull
git checkout (branch)
git pull . master

This merges it for you.

On Fri, Aug 16, 2019 at 6:13 PM waldroje <notifications@github.com> wrote:

> Sorry for all the commits, still have not figured out how to rebase
> properly.. I added another scheduler, I didn't run tests, wanted to see if
> it was something you were interested in first... I think it's the better
> version, as it evaluates all running trials at the same spot, and can stop
> a trial even if it is far ahead of slower trials based on past
> performance... it only evaluates once all have reached a multiple of the
> evaluation interval, and then looks at results in that window., so rather
> than evaluating each trial all the time, evaluates all running trials at
> the same time on a periodic schedule. This is the one I think I'm going to
> use, and if you think it's worthwhile, I'll fix the tests up.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ray-project/ray/pull/5402?email_source=notifications&email_token=ABCRZZODA3AIZP5E6QFY5A3QE5GCBA5CNFSM4IKFGW42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4QAGHQ#issuecomment-522191646>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCRZZLWFUAM7BHMTVVDTWDQE5GCBANCNFSM4IKFGW4Q>
> .
>

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/16360/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/17385/
Test FAILed.

ujvl commented:
Hi @waldroje, I cleaned things up and made some changes that should hopefully address your use-cases. 

> I think it's the better version, as it evaluates all running trials at the same spot, and can stop a trial even if it is far ahead of slower trials based on past performance... it only evaluates once all have reached a multiple of the evaluation interval, and then looks at results in that window., so rather than evaluating each trial all the time, evaluates all running trials at the same time on a periodic schedule. This is the one I think I'm going to use, and if you think it's worthwhile, I'll fix the tests up.

Instead of evaluating against the earliest common result which probably won't scale great, we can instead pause trials that don't have sufficient samples to eval against. When resumed they will likely have more peers. Setting `min_time_slice=0` would have the strictest behavior where at least `min_samples` trials will move more or less in lock-step. 

To alleviate checkpointing cost you can relax the constraint by setting `min_time_slice={some sufficiently large value}` so that each trial will run at least that long before potentially pausing.

This is similar to the param you introduced earlier (`eval_interval`) but the advantage over that is that we can keep the fine-grained evaluation while also preventing unnecessary checkpoint/restore overhead.

i think this should also now work well for your case where you have limited resources (eg: even if you can only run 2 trials simultaneously, if `min_samples > 2` then they would be preempted to evaluate other trials as well.

Let me know if these changes don't make sense.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/17386/
Test FAILed.

richardliaw commented on python/ray/tune/schedulers/median_stopping_rule.py:
```diff
@@ -77,24 +83,41 @@ def on_trial_result(self, trial_runner, trial, result):
         value by step `t` is strictly worse than the median of the running
         averages of all completed trials' objectives reported up to step `t`.
         """
-        if self._time_attr not in result or self._metric not in result:
```
let's keep this

richardliaw commented on python/ray/tune/schedulers/median_stopping_rule.py:
```diff
@@ -77,24 +83,41 @@ def on_trial_result(self, trial_runner, trial, result):
+            return TrialScheduler.CONTINUE
+
+        trials = self._trials_beyond_time(time)
+        trials.remove(trial)
```
is this needed?

ujvl replied:
It's more consistent with the Vizier paper's alg, i think they don't include it in the median computation. It also makes sense to me that if someone wants `k` samples to be used to calculate the median that that shouldn't include the current trial. Aside from that, whether or not it's really necessary is unclear to me.

waldroje commented:
Thanks for taking a look at this.. I've been buried lately and haven't had a chance to come back to it..

I think by applying _trials_beyond_time, that's a good solution, as that can replace the "lowest_common_interval" and all the overhead that went with it.. I think you just need to be able to apply a window in _running_mean .. i.e.
scoped_results = [r for r in results if r[self._time_attr] <= time][-self._running_window_size:]

fwiw, in my last implementation, that "eval_interval" only controlled the pace in which trials were compared... everything still stepped at the "time_attr".. "eval_interval" was units of "time_attr", and was my way of actually minimizing checkpointing... and yet keeping it separate from the rolling_window.. i.e. "evaluate every 25 training iterations, but only use results from the last 10 to make the stopping decision"

ujvl commented:
@waldroje 
> I think you just need to be able to apply a window in _running_mean .. i.e.
scoped_results = [r for r in results if r[self._time_attr] <= time][-self._running_window_size:]

I see, can you clarify why that would be useful for you——aside from the performance gain?

> fwiw, in my last implementation, that "eval_interval" only controlled the pace in which trials were compared

makes sense yeah

waldroje commented:
I may not be being clear here... as I don't think the window has any performance implications.. it purely impacts the value of the metric which is being used to stop/continue trials... if I'm reading the current code right, each trial that is returned from _trials_beyond_time has the mean([result[metric] for all_results] calculated, which is then used to calculate the median... at which point certain trials might be dropped, subject to _compare_op(median, best_result)... per [#5485] many (including myself), may not want the entire list of results to contribute to the mean calculation prior to calculating the median.... 

It's also possible some may want a window applied to _best_result as well, although for me, it's more likely if a trial at one point had good performance, you may want to keep it around, so I'm ok with using all results to calculate the best_result...

ujvl commented:
Got it, sorry I missed that. Will add a parameter to support it

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/17461/
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/17470/
Test FAILed.

richardliaw commented:
@ujvl Got this - 

```
Result for MyTrainableClass_2_height=74,width=46:
  date: 2019-10-07_22-36-45
  done: true
  episode_reward_mean: 73.9752420145566
  experiment_id: b62c7806ef93440994b0252d8896da8a
  hostname: C02TX1VXHTDD
  iterations_since_restore: 194
  node_ip: 128.32.38.223
  pid: 83185
  time_since_restore: 0.006340742111206055
  time_this_iter_s: 2.6226043701171875e-05
  time_total_s: 0.006569385528564453
  timestamp: 1570513005
  timesteps_since_restore: 0
  training_iteration: 200
  trial_id: 920fe8f4

2019-10-07 22:36:45,781	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=78.58281773787623 vs median res=45.401565668280085 at t=70
2019-10-07 22:36:45,791	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=78.98724643263539 vs median res=45.73199038189189 at t=71
2019-10-07 22:36:45,804	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=79.37825175166722 vs median res=46.05656940650776 at t=72
2019-10-07 22:36:45,813	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=79.75621605766295 vs median res=46.37540785284332 at t=73
2019-10-07 22:36:45,822	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=80.12151528243011 vs median res=46.68861004643098 at t=74
2019-10-07 22:36:45,831	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=80.47451869477129 vs median res=46.99627944105846 at t=75
2019-10-07 22:36:45,843	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=80.81558870116716 vs median res=47.298518539869384 at t=76
2019-10-07 22:36:45,855	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=81.14508067724128 vs median res=47.59542882373267 at t=77
2019-10-07 22:36:45,865	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=81.46334282803798 vs median res=47.887110686495724 at t=78
2019-10-07 22:36:45,875	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=81.77071607520125 vs median res=48.17366337674611 at t=79
2019-10-07 22:36:45,886	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=82.06753396920578 vs median res=48.455184945716404 at t=80
2019-10-07 22:36:45,900	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=82.354122624857 vs median res=48.73177220097815 at t=81
2019-10-07 22:36:45,912	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=82.63080067834562 vs median res=49.00352066558242 at t=82
2019-10-07 22:36:45,922	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=82.89787926421324 vs median res=49.27052454231624 at t=83
2019-10-07 22:36:45,934	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=83.15566201065687 vs median res=49.53287668275678 at t=84
2019-10-07 22:36:45,946	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=83.40444505167346 vs median res=49.79066856081755 at t=85
2019-10-07 22:36:45,959	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=83.64451705461799 vs median res=50.043990250493295 at t=86
2019-10-07 22:36:45,970	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=83.87615926182067 vs median res=50.29293040752309 at t=87
2019-10-07 22:36:45,978	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=84.09964554498119 vs median res=50.537576254703694 at t=88
2019-10-07 22:36:45,986	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=84.31524247112725 vs median res=50.77801357059745 at t=89
2019-10-07 22:36:45,993	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=84.52320937899515 vs median res=51.014326681391374 at t=90
2019-10-07 22:36:46,002	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=84.72379846475586 vs median res=51.246598455676136 at t=91
2019-10-07 22:36:46,013	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=84.91725487607691 vs median res=51.474910301925384 at t=92
2019-10-07 22:36:46,020	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.10381681357293 vs median res=51.69934216846716 at t=93
2019-10-07 22:36:46,027	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.28371563875952 vs median res=51.91997254575051 at t=94
2019-10-07 22:36:46,035	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.45717598768343 vs median res=52.13687847072099 at t=95
2019-10-07 22:36:46,043	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.62441588945948 vs median res=52.350135533129425 at t=96
2019-10-07 22:36:46,051	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.78564688899773 vs median res=52.55981788360812 at t=97
2019-10-07 22:36:46,058	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=85.94107417325745 vs median res=52.765998243358496 at t=98
2019-10-07 22:36:46,064	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.0908967004124 vs median res=52.9687479153035 at t=99
2019-10-07 22:36:46,070	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.23530733136027 vs median res=53.16813679656686 at t=100
2019-10-07 22:36:46,077	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.37449296305206 vs median res=53.36423339215006 at t=101
2019-10-07 22:36:46,083	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.50863466316086 vs median res=53.557104829685784 at t=102
2019-10-07 22:36:46,090	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.63790780564807 vs median res=53.7468168751545 at t=103
2019-10-07 22:36:46,099	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.76248220682352 vs median res=53.93343394945835 at t=104
2019-10-07 22:36:46,106	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.88252226153071 vs median res=54.117019145753254 at t=105
2019-10-07 22:36:46,112	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=86.99818707912218 vs median res=54.29763424744713 at t=106
2019-10-07 22:36:46,118	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.1096306189207 vs median res=54.47533974677834 at t=107
2019-10-07 22:36:46,124	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.21700182489178 vs median res=54.65019486389444 at t=108
2019-10-07 22:36:46,132	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.32044475927957 vs median res=54.82225756635715 at t=109
2019-10-07 22:36:46,140	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.42009873498473 vs median res=54.99158458900459 at t=110
2019-10-07 22:36:46,149	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.5160984464856 vs median res=55.1582314541072 at t=111
2019-10-07 22:36:46,157	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.60857409912701 vs median res=55.32225249175836 at t=112
2019-10-07 22:36:46,167	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.69765153662108 vs median res=55.48370086044517 at t=113
2019-10-07 22:36:46,176	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.78345236662359 vs median res=55.6426285677493 at t=114
2019-10-07 22:36:46,187	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.86609408426759 vs median res=55.79908649113179 at t=115
2019-10-07 22:36:46,193	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=87.94569019355121 vs median res=55.95312439875912 at t=116
2019-10-07 22:36:46,201	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.02235032649322 vs median res=56.10479097033177 at t=117
2019-10-07 22:36:46,210	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.09618035998207 vs median res=56.254133817879705 at t=118
2019-10-07 22:36:46,221	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.1672825302584 vs median res=56.40119950649208 at t=119
2019-10-07 22:36:46,227	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.23575554498198 vs median res=56.5460335749516 at t=120
2019-10-07 22:36:46,234	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.30169469284446 vs median res=56.688680556246595 at t=121
2019-10-07 22:36:46,240	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.36519195070011 vs median res=56.82918399793641 at t=122
2019-10-07 22:36:46,246	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.42633608819483 vs median res=56.96758648234798 at t=123
2019-10-07 22:36:46,252	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.48521276988203 vs median res=57.10392964658386 at t=124
2019-10-07 22:36:46,259	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.5419046548216 vs median res=57.23825420232371 at t=125
2019-10-07 22:36:46,269	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.59649149366467 vs median res=57.37059995540359 at t=126
2019-10-07 22:36:46,276	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.64905022323293 vs median res=57.501005825158714 at t=127
2019-10-07 22:36:46,282	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.69965505860657 vs median res=57.62950986351725 at t=128
2019-10-07 22:36:46,288	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.74837758273999 vs median res=57.75614927383431 at t=129
2019-10-07 22:36:46,294	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.79528683362872 vs median res=57.88096042945648 at t=130
2019-10-07 22:36:46,301	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.84044938905444 vs median res=58.00397889200868 at t=131
2019-10-07 22:36:46,308	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.88392944893901 vs median res=58.125239429396466 at t=132
2019-10-07 22:36:46,316	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.9257889153409 vs median res=58.24477603351765 at t=133
2019-10-07 22:36:46,323	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=88.9660874701303 vs median res=58.36262193767871 at t=134
2019-10-07 22:36:46,330	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.00488265038126 vs median res=58.47880963371174 at t=135
2019-10-07 22:36:46,336	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.0422299215213 vs median res=58.59337088878922 at t=136
2019-10-07 22:36:46,342	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.07818274828033 vs median res=58.70633676193417 at t=137
2019-10-07 22:36:46,349	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.11279266348251 vs median res=58.81773762022434 at t=138
2019-10-07 22:36:46,356	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.14610933472521 vs median res=58.92760315468962 at t=139
2019-10-07 22:36:46,363	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.17818062899072 vs median res=59.035962395902395 at t=140
2019-10-07 22:36:46,372	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.20905267523639 vs median res=59.1428437292613 at t=141
2019-10-07 22:36:46,378	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.23876992501032 vs median res=59.24827490996936 at t=142
2019-10-07 22:36:46,385	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.2673752111387 vs median res=59.35228307770766 at t=143
2019-10-07 22:36:46,391	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.29490980453257 vs median res=59.4548947710066 at t=144
2019-10-07 22:36:46,396	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.32141346916059 vs median res=59.55613594131658 at t=145
2019-10-07 22:36:46,403	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.34692451523505 vs median res=59.656031966780986 at t=146
2019-10-07 22:36:46,411	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.3714798506577 vs median res=59.75460766571407 at t=147
2019-10-07 22:36:46,419	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.39511503077212 vs median res=59.85188730978681 at t=148
2019-10-07 22:36:46,426	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.41786430646843 vs median res=59.94789463692431 at t=149
2019-10-07 22:36:46,433	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.4397606706862 vs median res=60.04265286391801 at t=150
2019-10-07 22:36:46,439	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.46083590336048 vs median res=60.136184698756736 at t=151
2019-10-07 22:36:46,445	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.48112061485534 vs median res=60.228512352680355 at t=152
2019-10-07 22:36:46,452	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.500644287929 vs median res=60.319657551960105 at t=153
2019-10-07 22:36:46,459	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.51943531827327 vs median res=60.40964154940991 at t=154
2019-10-07 22:36:46,467	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.53752105366998 vs median res=60.49848513563285 at t=155
2019-10-07 22:36:46,476	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.55492783180574 vs median res=60.586208650007215 at t=156
2019-10-07 22:36:46,484	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.57168101678597 vs median res=60.672831991416736 at t=157
2019-10-07 22:36:46,491	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.58780503438796 vs median res=60.758374628729264 at t=158
2019-10-07 22:36:46,499	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.60332340609217 vs median res=60.842855611028675 at t=159
2019-10-07 22:36:46,507	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.61825878193007 vs median res=60.92629357760452 at t=160
2019-10-07 22:36:46,514	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.63263297218548 vs median res=61.008706767704034 at t=161
2019-10-07 22:36:46,520	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.64646697798645 vs median res=61.09011303005113 at t=162
2019-10-07 22:36:46,527	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.6597810208228 vs median res=61.17052983213698 at t=163
2019-10-07 22:36:46,533	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.67259457102416 vs median res=61.24997426928694 at t=164
2019-10-07 22:36:46,539	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.68492637523241 vs median res=61.328463073508274 at t=165
2019-10-07 22:36:46,547	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.69679448290127 vs median res=61.40601262212319 at t=166
2019-10-07 22:36:46,553	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.70821627185524 vs median res=61.4826389461921 at t=167
2019-10-07 22:36:46,562	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.71920847293896 vs median res=61.558357738731125 at t=168
2019-10-07 22:36:46,569	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.72978719378733 vs median res=61.633184362728805 at t=169
2019-10-07 22:36:46,575	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.73996794174593 vs median res=61.70713385896603 at t=170
2019-10-07 22:36:46,581	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.74976564597011 vs median res=61.780220953643905 at t=171
2019-10-07 22:36:46,587	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.75919467873088 vs median res=61.852460065823664 at t=172
2019-10-07 22:36:46,594	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.76826887595435 vs median res=61.92386531468302 at t=173
2019-10-07 22:36:46,602	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.77700155702088 vs median res=61.99445052659311 at t=174
2019-10-07 22:36:46,609	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.78540554384944 vs median res=62.06422924202029 at t=175
2019-10-07 22:36:46,617	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.79349317929173 vs median res=62.13321472225664 at t=176
2019-10-07 22:36:46,623	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.80127634485989 vs median res=62.20141995598355 at t=177
2019-10-07 22:36:46,629	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.80876647781098 vs median res=62.26885766567206 at t=178
2019-10-07 22:36:46,635	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.81597458761054 vs median res=62.3355403138239 at t=179
2019-10-07 22:36:46,641	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.82291127179698 vs median res=62.40148010905727 at t=180
2019-10-07 22:36:46,649	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.82958673126781 vs median res=62.466689012040774 at t=181
2019-10-07 22:36:46,655	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.83601078500789 vs median res=62.53117874127946 at t=182
2019-10-07 22:36:46,663	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.84219288427953 vs median res=62.594960778756494 at t=183
2019-10-07 22:36:46,670	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.84814212629342 vs median res=62.658046375433905 at t=184
2019-10-07 22:36:46,676	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.85386726737865 vs median res=62.7204465566161 at t=185
2019-10-07 22:36:46,682	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.85937673566983 vs median res=62.78217212717932 at t=186
2019-10-07 22:36:46,688	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.86467864332833 vs median res=62.843233676670515 at t=187
2019-10-07 22:36:46,695	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.86978079831438 vs median res=62.90364158427885 at t=188
2019-10-07 22:36:46,703	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.87469071572605 vs median res=62.96340602368295 at t=189
2019-10-07 22:36:46,712	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.87941562872058 vs median res=63.02253696777722 at t=190
2019-10-07 22:36:46,719	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.88396249903325 vs median res=63.081044193280114 at t=191
2019-10-07 22:36:46,725	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.88833802710806 vs median res=63.1389372852272 at t=192
2019-10-07 22:36:46,731	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.89254866185432 vs median res=63.196225641352385 at t=193
2019-10-07 22:36:46,737	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.89660061004273 vs median res=63.252918476359746 at t=194
2019-10-07 22:36:46,746	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.90049984535375 vs median res=63.309024826088795 at t=195
2019-10-07 22:36:46,754	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.90425211709136 vs median res=63.36455355157613 at t=196
2019-10-07 22:36:46,762	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.90786295857365 vs median res=63.41951334301596 at t=197
2019-10-07 22:36:46,769	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.91133769521262 vs median res=63.47391272362192 at t=198
2019-10-07 22:36:46,775	INFO median_stopping_rule.py:121 -- Trial MyTrainableClass_6_height=90,width=52 best res=89.9146814522943 vs median res=63.52776005339321 at t=199
```

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/17548/
Test FAILed.
